swift/Chat/views/MessageView.swift

```diff
@@ -0,0 +1,275 @@
+            // the initial scroll flash by
+            uiViewController.messagesCollectionView.scrollToLastItem(animated: self.initialized)
+            self.initialized = true
+        }
+    }
+
+    private func isLastSectionVisible(_ uiViewController: MessagesViewController) -> Bool {
+        guard !$client.messages.isEmpty else { return false }
+        let lastIndexPath = IndexPath(item: 0, section: $client.messages.count - 1)
+        return uiViewController.messagesCollectionView.indexPathsForVisibleItems.contains(lastIndexPath)
+    }
+}
+
+extension MessagesView.Coordinator: MessagesDataSource {
+    func currentSender() -> SenderType {
+        return client.currentUser
+    }
+
+    func messageForItem(at indexPath: IndexPath, in _: MessagesCollectionView) -> MessageType {
+        return client.messages[indexPath.section]
+    }
+
+    func numberOfSections(in _: MessagesCollectionView) -> Int {
+        return client.messages.count
+    }
+
+    func messageTopLabelAttributedText(for message: MessageType, at _: IndexPath) -> NSAttributedString? {
+        let name = message.sender.displayName
+        let font = UIFont.preferredFont(forTextStyle: .caption1)
+        return NSAttributedString(string: name,
+                                  attributes: [NSAttributedString.Key.font: font])
+    }
+
+    func messageBottomLabelAttributedText(for message: MessageType, at _: IndexPath) -> NSAttributedString? {
+        let dateString = formatter.string(from: message.sentDate)
+        let font = UIFont.preferredFont(forTextStyle: .caption2)
+        return NSAttributedString(string: dateString,
+                                  attributes: [NSAttributedString.Key.font: font])
+    }
+
+    func messageTimestampLabelAttributedText(for message: MessageType, at _: IndexPath) -> NSAttributedString? {
+        let sentDate = message.sentDate
+        let sentDateString = MessageKitDateFormatter.shared.string(from: sentDate)
+        let timeLabelFont: UIFont = .boldSystemFont(ofSize: 10)
+        let timeLabelColor: UIColor = .systemGray
+        return NSAttributedString(string: sentDateString,
+                                  attributes: [NSAttributedString.Key.font: timeLabelFont,
+                                               NSAttributedString.Key.foregroundColor: timeLabelColor])
+    }
+}
+
+extension MessagesView.Coordinator: InputBarAccessoryViewDelegate {
+    func inputBar(_ inputBar: InputBarAccessoryView, didPressSendButtonWith _: String) {
+        let text = inputBar.inputTextView.text ?? ""
+        inputBar.inputTextView.text = String()
+        inputBar.invalidatePlugins()
+        inputBar.sendButton.startAnimating()
+        inputBar.inputTextView.placeholder = "Sending"
+        DispatchQueue.global(qos: .default).async {
+            self.client.session?.sendAsync(text).catch {
+                print($0)
+                // self.exception(err: $0)
```

Comment:
Why is this commented out, how do we handle the send error? What about the print statement is it temporary?

Reply:
Yes the print statement was temporary. Just added proper displaying of error. Will commit changes that address other comments shortly too!